eggplant/profiles/views.py

```diff
@@ -103,12 +105,17 @@ def get_initial(self):
 
     def form_valid(self, form):
         user_id = self.request.user.id
+        photo = self.request.FILES.get('photo')
+
         User.objects.filter(id=user_id)\
             .update(first_name=form.cleaned_data['first_name'],
                     last_name=form.cleaned_data['last_name'])
         del form.cleaned_data['first_name']
         del form.cleaned_data['last_name']
+        del form.cleaned_data['photo']
```

Comment:
Maybe this line is problematic? Actually, why are these deleted from the cleaned_data object? I think it should be part of forms.cleaned_data which is set for the UserProfile. You could use `self.object.field = form.cleaned_data[...]` and then `self.object.save()`....